lib/internal/test_runner/tap_stream.js

```diff
@@ -85,22 +89,23 @@ class TapStream extends Readable {
   }
 
   diagnostic(indent, message) {
+    this.emit('test:diagnostic', message);
     this.#tryPush(`${indent}# ${tapEscape(message)}\n`);
   }
 
   version() {
     this.#tryPush('TAP version 13\n');
   }
 
-  #test(indent, testNumber, status, description, directive) {
+  #test(indent, testNumber, status, name, directive = kEmptyObject) {
     let line = `${indent}${status} ${testNumber}`;
 
-    if (description) {
-      line += ` ${tapEscape(description)}`;
+    if (name) {
+      line += ` ${tapEscape(`- ${name}`)}`;
     }
 
-    if (directive) {
-      line += ` # ${directive}`;
+    for (const key in directive) {
+      line += ` # ${key.toUpperCase()}${directive[key] ? ` ${tapEscape(directive[key])}` : ''}`;
```

Comment:
this will enumerate Object.prototype, so i think you want to instead do something like
```suggestion
    line += ArrayPrototypeJoin(ArrayPrototypeMap(ObjectKeys(directive, (key) => (
      ` # ${key.toUpperCase()}${directive[key] ? ` ${tapEscape(directive[key])}` : ''}`
    )), '');
```